server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -445,16 +445,18 @@ public static void updateRepositoryUuidInMetadata(
             new ClusterStateUpdateTask() {
                 @Override
                 public ClusterState execute(ClusterState currentState) {
-                    final RepositoriesMetadata currentReposMetadata = RepositoriesMetadata.get(currentState);
+                    final var project = currentState.metadata().getDefaultProject();
+                    final RepositoriesMetadata currentReposMetadata = RepositoriesMetadata.get(project);
 
                     final RepositoryMetadata repositoryMetadata = currentReposMetadata.repository(repositoryName);
                     if (repositoryMetadata == null || repositoryMetadata.uuid().equals(repositoryUuid)) {
                         return currentState;
                     } else {
                         final RepositoriesMetadata newReposMetadata = currentReposMetadata.withUuid(repositoryName, repositoryUuid);
-                        final Metadata.Builder metadata = Metadata.builder(currentState.metadata())
-                            .putDefaultProjectCustom(RepositoriesMetadata.TYPE, newReposMetadata);
-                        return ClusterState.builder(currentState).metadata(metadata).build();
+                        return currentState.copyAndUpdateProject(
+                            project.id(),
+                            builder -> builder.putCustom(RepositoriesMetadata.TYPE, newReposMetadata)
+                        );
```

Comment:
Sorry to drag on this. But can we please change this to use the existing style as well?

Reply:
Haha no worries. Thanks for reviewing!